### ISSUE TITLE
False positive: UnnecessaryParentheses in selector expression of switch expressions

### ISSUE THREAD

**Issue by anthonyvdotbe** [google/error-prone#1647]
### Description of the problem / feature request:

A false positive for `UnnecessaryParentheses` is reported when the selector expression of a switch expression is, for example, a method invocation.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

````java
class Foo {
    boolean bar() {
        return switch(baz()) {
            case 42 -> true;
            default -> false;
        };
    }

    int baz() {
        return 42;
    }
}
````

### What version of Error Prone are you using?

2.4.0

###  Have you found anything relevant by searching the web?

No.

**Comment by HonoluluHenk** [comment]
For me, also happens on a simple switch expression over an enum:

```java
enum Foo {
	FOO,
	BAR
}

public Foo whatNext(Foo foo) {
	var next = switch (foo) {
		case FOO -> Foo.BAR;
		case BAR -> Foo.FOO;
	};
	return next;
}
```

Using Adopt OpenJDK 14

**Comment by codylerum** [comment]
False positive happens on this simple case as well

```
 public Boolean isMultiLocationTrunk() {
        return switch (multiLocationTrunkInput) {
            case 0 -> null;
            case 1 -> true;
            case 2 -> false;
            default -> null;
        };
    }
```

**Comment by dododge** [comment]
Also running into this with 2.6.0.

**Comment by codylerum** [comment]
@cushon I'm working on a fix for this but I'm unable to execute the `UnnecessaryParenthesesTest.java` class in Intellij and debug it as I hit lots of duplicate class build errors.

```
V:\IdeaProjects\error-prone\core\src\test\java\com\google\errorprone\bugpatterns\flogger\testdata\FloggerRedundantIsEnabledPositiveCases_expected.java:22
java: duplicate class: com.google.errorprone.bugpatterns.flogger.testdata.FloggerRedundantIsEnabledPositiveCases
```

Any tricks there for setting up the environment or excludes? The test I'm trying to add to `UnnecessaryParenthesesTest` is below which should pass without raising any errors

``` @Test
  public void switchExpression() {
    helper
            .addSourceLines(
                    "Test.java",
                    "class Test {",
                    "  public boolean match(String value) {",
                    "    return switch (value) {",
                    "    case \"true\" -> true;",
                    "    case \"false\" -> false;",
                    "    default -> throw new RuntimeException(\"Unable to match\");",
                    "    };",
                    "  }",
                    "}")
            .expectResult(Main.Result.OK)
            .doTest();
  }
```

**Comment by cushon** [comment]
> Any tricks there for setting up the environment or excludes?

@codylerum you might need to add the various `/testdata/` directories to excludes. I also use IntelliJ and can run that test.

**Comment by codylerum** [comment]
For anyone else who finds this problem I was able to get to the point of debugging an individual test by selecting the delegate to maven in the runner settings and checking the skip tests.

**Comment by codylerum** [comment]
Thanks @cushon I was able to get debugging running for an individual test in intellij and I think found the solution. PR linked.
